tools: add some faq for br, cdc and binlog (#4592)
pingcap/docs: ticdc/troubleshoot-ticdc.md

    {{< copyable "shell-regular" >}}

    ```shell
    cdc cli changefeed create --sink-uri="mysql://root@127.0.0.1:3306/?time-zone=CST"
    ```

    > **Note:**
    >
    > CST might be an abbreviation for the following four different time zones:、
    >
    > + Central Standard Time (USA) UT-6:00
    > + Central Standard Time (Australia) UT+9:30
    > + China Standard Time UT+8:00
    > + Cuba Standard Time UT-4:00
    >
    > In China, CST usually means China Standard Time, and pay attention to the discrimination when using it.

## How to understand the relationship between TiCDC time zone and upstream and downstream databases time zones?

||Upstream time zone| TiCDC time zone|Downstream time zone|
| :-: | :-: | :-: | :-: |

> | Configuration method | See [Time Zone Support](/configure-time-zone.md) | The `--tz` parameter when starting ticdc server | The `time-zone` parameter in `sink-uri` |

Reviewer: ```suggestion
| Configuration method | See [Time Zone Support](/configure-time-zone.md) | Configured using the `--tz` parameter when you start the TiCDC server |  Configured using the `time-zone` parameter in `sink-uri` |
```

> | Description | The time zone of the upstream TiDB affects DML operations of timestamp type and DDL operations related to timestamp type columns.| TiCDC assumes that the upstream TiDB's time zone is the same as the TiCDC time zone configuration, and performs related operations on the timestamp column.| The downstream MySQL processes the timestamp contained in the DML and DDL operations according to the downstream time zone settings.|

Reviewer: ```suggestion
| Description | The time zone of the upstream TiDB, which affects DML operations of the timestamp type and DDL operations related to timestamp type columns.| TiCDC assumes that the upstream TiDB's time zone is the same as the TiCDC time zone configuration, and performs related operations on the timestamp column.| The downstream MySQL processes the timestamp in the DML and DDL operations according to the downstream time zone setting.|
```

 > **Note:**
 >

>  > Be cautious when you set the time zone of the TiCDC server, because the time zone will be used for the conversion of time type. You should keep the upstream time zone, TiCDC time zone and downstream time zone consistent.
>  > The TiCDC server chooses its time zone in the following priority:

Reviewer: ```suggestion
 > Be careful when you set the time zone of the TiCDC server, because this time zone is used for converting the time type. Keep the upstream time zone, TiCDC time zone, and the downstream time zone consistent. The TiCDC server chooses its time zone in the following priority:
```

    {{< copyable "shell-regular" >}}

    ```shell
    cdc cli changefeed create --sink-uri="mysql://root@127.0.0.1:3306/?time-zone=CST"
    ```

    > **Note:**
    >
    > CST might be an abbreviation for the following four different time zones:
    >
    > + Central Standard Time (USA) UT-6:00
    > + Central Standard Time (Australia) UT+9:30
    > + China Standard Time UT+8:00
    > + Cuba Standard Time UT-4:00
    >
    > In China, CST usually stands for China Standard Time.

> ## How to understand the relationship of the TiCDC time zone with the time zones of the upstream/downstream databases?

Reviewer: ```suggestion
## How to understand the relationship between the TiCDC time zone and the time zones of the upstream/downstream databases?
```

||Upstream time zone| TiCDC time zone|Downstream time zone|
| :-: | :-: | :-: | :-: |
| Configuration method | See [Time Zone Support](/configure-time-zone.md) | Configured using the `--tz` parameter when you start the TiCDC server |  Configured using the `time-zone` parameter in `sink-uri` |
| Description | The time zone of the upstream TiDB, which affects DML operations of the timestamp type and DDL operations related to timestamp type columns.| TiCDC assumes that the upstream TiDB's time zone is the same as the TiCDC time zone configuration, and performs related operations on the timestamp column.| The downstream MySQL processes the timestamp in the DML and DDL operations according to the downstream time zone setting.|

 > **Note:**
 >
 > Be careful when you set the time zone of the TiCDC server, because this time zone is used for converting the time type. Keep the upstream time zone, TiCDC time zone, and the downstream time zone consistent. The TiCDC server chooses its time zone in the following priority:
 >

>  > - TiCDC first uses the time zone specified by `--tz`.
>  > - When `--tz` is not available, TiCDC tries to read the time zone set by the `TZ` environment variable.
>  > - When the `TZ` environment variable is not available, TiCDC uses the default time zone of the machine.

Reviewer: ```suggestion
 > - TiCDC first uses the time zone specified using `--tz`.
 > - When `--tz` is not available, TiCDC tries to read the time zone set using the `TZ` environment variable.
 > - When the `TZ` environment variable is not available, TiCDC uses the default time zone of the machine.
```

* `DELETE` event only contains the `"d"` field

For more information, refer to [Open protocol Row Changed Event format](/ticdc/ticdc-open-protocol.md#row-changed-event).

## Does TiCDC support replicating large transactions? Is there any risk?

TiCDC provides partial support for large transactions (more than 5 GB in size). Depending on different scenarios, the following risks might exist:

> - When the internal processing capacity of TiCDC is insufficient, the replication task error `ErrBufferReachLimit` might occur.
> - When TiCDC internal processing capacity is insufficient or TiCDC downstream throughput capacity is insufficient, out of memory (OOM) might occur.

Reviewer: ```suggestion
- When TiCDC's internal processing capacity is insufficient, the replication task error `ErrBufferReachLimit` might occur.
- When TiCDC's internal processing capacity is insufficient or the throughput capacity of TiCDC's downstream is insufficient, out of memory (OOM) might occur.
```

> When you encounter the above errors, it is recommended to use BR for incremental restore of incremental data containing large transactions. The specific operations are as follows:

Reviewer: ```suggestion
If you encounter errors above, it is recommended to use BR for incrementally restoring data in large transactions. The detailed operations are as follows:
```

1. Record the `checkpoint-ts` of the changefeed that is terminated due to large transactions, use this TSO as the `--lastbackupts` of the BR incremental backup, and execute [Back up incremental data](/br/backup-and-restore-tool.md#back-up-incremental-data).

> 2. After backing up the incremental data, you can find a log similar to `["Full backup Failed summary : total backup ranges: 0, total success: 0, total failed: 0"] [BackupTS=421758868510212097]` in the BR log output and record the `BackupTS` in this log.

Reviewer: ```suggestion
2. After backing up the incremental data, you can find a log similar to `["Full backup Failed summary : total backup ranges: 0, total success: 0, total failed: 0"] [BackupTS=421758868510212097]` in the BR log output. Record the `BackupTS` in this log.
```

> 3. Execute [Restore incremental data](/br/backup-and-restore-tool.md#restore-incremental-data).
> 4. Create a new changefeed and start the replication task from `BackupTS`
> 5. Delete old changefeed.

Reviewer: ```suggestion
3. [Restore the incremental data ](/br/backup-and-restore-tool.md#restore-incremental-data).
4. Create a new changefeed and start the replication task from `BackupTS`.
5. Delete the old changefeed.
```

> ## If the downstream of changefeed is a database similar to MySQL, TiCDC executes a long DDL statement and blocks all other changefeed. What should I do?

Reviewer: ```suggestion
## When the downstream of a changefeed is a database similar to MySQL and TiCDC executes a long DDL statement, all other changefeeds are blocked. How should I handle the issue?
```

For more information, refer to [Open protocol Row Changed Event format](/ticdc/ticdc-open-protocol.md#row-changed-event).

## Does TiCDC support replicating large transactions? Is there any risk?

TiCDC provides partial support for large transactions (more than 5 GB in size). Depending on different scenarios, the following risks might exist:

- When TiCDC's internal processing capacity is insufficient, the replication task error `ErrBufferReachLimit` might occur.
- When TiCDC's internal processing capacity is insufficient or the throughput capacity of TiCDC's downstream is insufficient, out of memory (OOM) might occur.

> If you encounter errors above, it is recommended to use BR for incrementally restoring data in large transactions. The detailed operations are as follows:

Reviewer: ```suggestion
If you encounter an error above, it is recommended to use BR to restore the incremental data of large transactions. The detailed operations are as follows:
```

> 1. Record the `checkpoint-ts` of the changefeed that is terminated due to large transactions, use this TSO as the `--lastbackupts` of the BR incremental backup, and execute [Back up incremental data](/br/backup-and-restore-tool.md#back-up-incremental-data).

Reviewer: ```suggestion
1. Record the `checkpoint-ts` of the changefeed that is terminated due to large transactions, use this TSO as the `--lastbackupts` of the BR incremental backup, and execute [incremental data backup](/br/backup-and-restore-tool.md#back-up-incremental-data).
```

> 2. After backing up the incremental data, you can find a log similar to `["Full backup Failed summary : total backup ranges: 0, total success: 0, total failed: 0"] [BackupTS=421758868510212097]` in the BR log output. Record the `BackupTS` in this log.

Reviewer: ```suggestion
2. After backing up the incremental data, you can find a log record similar to `["Full backup Failed summary : total backup ranges: 0, total success: 0, total failed: 0"] [BackupTS=421758868510212097]` in the BR log output. Record the `BackupTS` in this log.
```

3. [Restore the incremental data](/br/backup-and-restore-tool.md#restore-incremental-data).
4. Create a new changefeed and start the replication task from `BackupTS`.
5. Delete the old changefeed.

> ## When the downstream of a changefeed is a database similar to MySQL and TiCDC executes a long DDL statement, all other changefeeds are blocked. How should I handle the issue?

Reviewer: ```suggestion
## When the downstream of a changefeed is a database similar to MySQL and TiCDC executes a time-consuming DDL statement, all other changefeeds are blocked. How should I handle the issue?
```

> 1. Pause the execution of the changefeed that contains the time-consuming DDL statement. Then, you can see that other changefeeds are no longer blocked.

Reviewer: ```suggestion
1. Pause the execution of the changefeed that contains the time-consuming DDL statement. Then you can see that other changefeeds are no longer blocked.
```

> 1. Firstly, suspend the execution of the changefeed of the long DDL. At this point, you can find that after you suspend this changefeed, other changefeeds are no longer blocked.

Reviewer: ```suggestion
1. Pause the execution of the changefeed that contains the time-consuming DDL statement. Then, you can see that other changefeeds are no longer blocked.
```

> 2. Search for the `apply job` field in the TiCDC log and confirm the time-consuming DDL `StartTs`.

Reviewer: ```suggestion
2. Search for the `apply job` field in the TiCDC log and confirm the `StartTs` of the time-consuming DDL statement.
```

> 3. Execute the DDL statement downstream manually, and perform the following operations after execution.

Reviewer: ```suggestion
3. Manually execute the DDL statement in the downstream. After the execution finishes, go on performing the following operations.
```

4. Modify the changefeed configuration and add the above `StartTs` to the `ignore-txn-start-ts` configuration item.

> 5. Restore a suspended changefeed.

Reviewer: ```suggestion
5. Resume the paused changefeed.
```

> ## After I upgrade TiCDC cluster to v4.0.8, changefeed reports the error `[CDC:ErrKafkaInvalidConfig]Canal requires old value to be enabled`

Reviewer: ```suggestion
## After I upgrade the TiCDC cluster to v4.0.8, the `[CDC:ErrKafkaInvalidConfig]Canal requires old value to be enabled` error is reported when I execute a changefeed.
```

> ## After I upgrade the TiCDC cluster to v4.0.8, the `[CDC:ErrKafkaInvalidConfig]Canal requires old value to be enabled` error is reported when I execute a changefeed.

Reviewer: ```suggestion
## After I upgrade the TiCDC cluster to v4.0.8, the `[CDC:ErrKafkaInvalidConfig]Canal requires old value to be enabled` error is reported when I execute a changefeed
```

> Since v4.0.8, if changefeed uses canal or canal-json protocol output, TiCDC checks whether you enable the Old Value feature. If you disable it, an error is reported. You can solve the problem by following the steps below:

Reviewer: ```suggestion
Since v4.0.8, if the `canal` or `canal-json` protocol is used for output in a changefeed, TiCDC checks whether the old value feature is also enabled. If you disable the old value feature, this error is reported. To fix the error, take the following steps:
```

1. Set the value of `enable-old-value` in the changefeed configuration file to `true`.

> 2. Use `cdc cli changefeed update` to update the original changefeed configuration.

Reviewer: ```suggestion
2. Execute `cdc cli changefeed update` to update the original changefeed configuration.
```

    {{< copyable "shell-regular" >}}

    ```shell
    cdc cli changefeed update -c test-cf --sink-uri="mysql://127.0.0.1:3306/?max-txn-row=20&worker-number=8" --config=changefeed.toml
    ```

> 3. Use `cdc cli changfeed resume` to restore replication tasks

Reviewer: ```suggestion
3. Execute `cdc cli changfeed resume` to resume the replication task.
```
